go/cmd/dolt/commands/diff.go

```diff
@@ -281,41 +285,52 @@ func parseDiffArgs(ctx context.Context, dEnv *env.DoltEnv, apr *argparser.ArgPar
 	return from, to, dArgs, nil
 }
 
-func getDiffRoots(ctx context.Context, dEnv *env.DoltEnv, args []string) (from, to *doltdb.RootValue, leftover []string, err error) {
-	headRoot, err := dEnv.StagedRoot(ctx)
+func getDiffRoots(ctx context.Context, dEnv *env.DoltEnv, args []string, isCached bool) (from, to *doltdb.RootValue, leftover []string, err error) {
+	stagedRoot, err := dEnv.StagedRoot(ctx)
 	workingRoot, err := dEnv.WorkingRootWithDocs(ctx)
+	headRoot, err := dEnv.HeadRoot(ctx)
 	if err != nil {
 		return nil, nil, nil, err
 	}
 
+	// default unstaged roots
+	fromRoot := stagedRoot
+	toRoot := workingRoot
```

Comment:
It's unnecessary to introduce the variables fromRoot and toRoot.  You can just use:

``` go
from = stagedRoot
to = workingRoot
```

this will mean that all the places below where you do `to = toRoot` or `from = fromRoot` won't be needed.